### PR TITLE
Handle walletconnect switch and add signing accounts

### LINKED ISSUE #1947: Handle WalletConnect switch and add signing accounts
### Description

A user can request to sign a transaction from Lisk App, where a signing account is not the same as Lisk Mobile current account or the signing account might not exists.

### Acceptance Criteria

- Allow users to [switch](https://www.figma.com/file/jjuNNU9XxpeL57P25H1qri/Mobile-Prototype?type=design&node-id=3745-61000&mode=design&t=je3AxnAihlxkDLTs-4) from current account to signing account on signature requests.
- Inform users to [add signing account](https://www.figma.com/file/jjuNNU9XxpeL57P25H1qri/Mobile-Prototype?type=design&node-id=3745-62759&mode=design&t=je3AxnAihlxkDLTs-4) if it doesn't exists in Lisk Mobile.
- Ensure Invalid/unsupported transaction and rejected.

### PR DESCRIPTION
**Pull request by Balanced02**
### What was the problem?

This PR resolves #1947 

### How was it solved?

Implement switching of accounts

### How was it tested?
https://github.com/LiskHQ/lisk-mobile/assets/28750984/2d73a25c-93d0-4c21-9644-c607ebb57a70

https://github.com/LiskHQ/lisk-mobile/assets/28750984/bdc53159-51c6-4bee-b700-6fbfaef5b03b

### CODE REVIEW

File: src/modules/BlockchainApplication/components/ExternalApplicationSignatureRequest/ExternalAppSignatureRequestNotification.js [hunk 1]
```diff
@@ -19,9 +22,19 @@ export default function ExternalAppSignatureRequestNotification({
...
+  switchAccount,
+  isCurrentAccount,
+  isAccountAdded,
+  navigation,
 }) {
   const { accounts } = useAccounts();
 
+  const addAccount = () => {
+    onCancel();
+    navigation.navigate('AccountsManagerScreen');
```
Comment (ClementeSerrano) [inline src/modules/BlockchainApplication/components/ExternalApplicationSignatureRequest/ExternalAppSignatureRequestNotification.js:35]: shouldn't we redirect to the AddAccount screen?

File: src/navigation/components/Tabs/index.js [hunk 2]
```diff
@@ -14,8 +15,9 @@ const Tab = createBottomTabNavigator();
 
 export default function Tabs() {
   const { isOpen: modalOpen } = useContext(ModalContext);
+  const navigation = useNavigation();
 
-  useExternalApplicationSignatureRequest();
+  useExternalApplicationSignatureRequest(navigation);
```
Comment (ClementeSerrano) [inline src/navigation/components/Tabs/index.js:20]: i think we can use navigation inside the hook

**Review by ClementeSerrano — APPROVED** [review]

File: src/modules/BlockchainApplication/components/ExternalApplicationSignatureRequest/ExternalAppSignatureRequestNotification.js [hunk 3]
```diff
@@ -81,22 +94,71 @@ export default function ExternalAppSignatureRequestNotification({
...
+              {i18next.t('application.externalApplicationSignatureRequest.errors.accountNotAdded1')}
+              <B style={[styles.theme.description]}>{stringShortener(signingAddress, 7, 4)}</B>
+              {i18next.t('application.externalApplicationSignatureRequest.errors.accountNotAdded2')}
+            </P>
+          }
+          variant="warning"
+        />
+      )}
+
+      {!isCurrentAccount && isAccountAdded && (
```
Comment (ManuGowda) [inline src/modules/BlockchainApplication/components/ExternalApplicationSignatureRequest/ExternalAppSignatureRequestNotification.js:133]: This and link 104 `{isAccountAdded && !isCurrentAccount && (` is the same condition? 🤔

File: src/modules/BlockchainApplication/components/ExternalApplicationSignatureRequest/index.js [hunk 4]
```diff
@@ -79,6 +83,24 @@ export default function ExternalApplicationSignatureRequest({ onCancel }) {
...
   };
 
+  const request = event?.meta?.params.request;
+
+  useEffect(() => {
+    if (request) {
+      try {
+        const { payload, schema } = request.params;
+        let transactionObj;
+        transactionObj = decodeTransaction(Buffer.from(payload, 'hex'), schema);
```
Comment (ManuGowda) [inline src/modules/BlockchainApplication/components/ExternalApplicationSignatureRequest/index.js:93]: add `validator.validate(schema, transactionObj.params)`

**Pull request merged**